### PR TITLE
 add the functions for the dailyGoal + functions for the achievements…

### LINKED ISSUE #97: Link the dailyUserGoals interface and element to the database + add testify it correctly modify it

### PR DESCRIPTION
**Pull request by HugoHof**
This PR contains:
- a function in the database to add a daily goal in the list of daily goal that the user has realized.
- adding the listDailyGoal argument to store the list of daily goal in the database 
- modify the getUser functions in the database to retrieve the dailyGoal list of the user from the database
- adding a getter and updateListDailyGoal function to get the daily goal list and add a daily goal to the daily goal list 
- adding a function in the database to update the achievements (total time run by user, total activity time, total number of paths draw by the user) with the value of a drawing activity at the end of this one( time of the drawing activity + distance of the drawing + 1 path)
- adding the achievements in the user model.
- add set and update function to get or update the achievement in the user model. 
- update the getUser functions in the database to retrieve the achievement from the database.-add the initialization of the achievement in the initUserProfile function in the database.

Thank you for your review
Don't hesitate if you have any question
Close #97

### CODE REVIEW

File: app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt [hunk 1]
```diff
@@ -138,16 +139,26 @@ class FireDatabase : Database() {
...
+        userData.put(currentDistanceGoalFile, userModel.getCurrentDistanceGoal())
+        userData.put(currentActivityTimeGoalFile, userModel.getCurrentActivityTime())
+        userData.put(currentNOfPathsGoalFile, userModel.getCurrentNumberOfPathsGoal())
+
+        updateUserData(userData).thenApply {
+            initUserAchievement().thenAccept { isInit ->
+                if(isInit){
+                    future.complete(true)
+                }
+                future.completeExceptionally(Exception("The user profile was not correctly initiate in the database !"))
```
Comment (HoLuDyEr) [inline app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt:158]: Detail: typo
The user profile was not correctly initiate in the database ! -> The user profile was not correctly initialized in the database !

File: app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt [hunk 2]
```diff
@@ -250,6 +261,77 @@ class FireDatabase : Database() {
...
+            newAchievement.put(totalDistanceFile, pastAchievements.get(totalDistanceFile) as Double + distanceDrawing)
+            newAchievement.put(totalActivityTimeFile, pastAchievements.get(totalActivityTimeFile) as Double + activityTimeDrawing)
+            newAchievement.put(totalNbOfPathsFile, pastAchievements.get(totalNbOfPathsFile) as Int + 1)
+            val userId = getUserId()
+            if (userId == null) {
+                future.completeExceptionally(java.lang.Error("The userId can't be null !"))
+            } else {
+                accessUserAccountFile(userId).child(achievementsFile).updateChildren(newAchievement)
+                    .addOnSuccessListener { future.complete(Unit) }
+                    .addOnFailureListener{ it }
```
Comment (HoLuDyEr) [inline app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt:329]: same remark as above

**Review by HoLuDyEr — CHANGES_REQUESTED** [review]
Overall it is a good PR, There are only some small details such as return the same futures type and returning the errors.

**Comment by HugoHof** [comment]
> Thank you for the changes it is ready to be merged

Thank you for your review

**Pull request merged**